src/azure-cli/azure/cli/command_modules/acr/custom.py

```diff
@@ -194,12 +194,11 @@ def acr_login(cmd,
             p.wait()
         else:
             if b'--password-stdin' in stderr:
-                errors = [err for err in stderr.decode().split('\n') if '--password-stdin' not in err]
-                stderr = '\n'.join(errors).encode()
+                errors = [err for err in stderr.decode().split('\n') if '--password-stdin' not in err and err != '']
 
-            import sys
-            output = getattr(sys.stderr, 'buffer', sys.stderr)
-            output.write(stderr)
+                if errors:
+                    stderr = '\n'.join(errors)
```

Comment:
@djyou, before I merge this change, want to confirm that it was not on purpose we void the exit code of 1

Reply:
We should raise regardless if ever reaching here, try the following
```
            if b'--password-stdin' in stderr:
                errors = ...
                stderr = '\n'.join(errors) if errors else 'some generic message'
            raise CLIError('docker: ' + stderr)
```
nit: not sure about the docker prefix in message but will let you decide.

Reply:
> We should raise regardless if ever reaching here, try the following
> 
> ```
>             if b'--password-stdin' in stderr:
>                 errors = ...
>                 stderr = '\n'.join(errors) if errors else 'some generic message'
>             raise CLIError('docker: ' + stderr)
> ```


I think the idea here is that, if the error is just "--password-stdin..." (meaning the login has succeed), the function should return normally and we should not show any error messages to user. if there are errors other than "--password-stdin", then we should raise

Reply:
What do we print if --password-stdin is not in stderr?

Reply:
Should raise error. Updated in latest commit.

Reply:
LGTM